pkg/odo/cli/component/create.go

```diff
@@ -38,280 +41,276 @@ var (
 	cpu              string
```

Comment:
I felt `CreateOptions` is supposed to be a container of attributes that the create command requires
but in this case, many of these fields like the componentBinary/componentGit/componentLocal, only one of them is required for create. So, I kept them separate... And only have those attrs that create actually requires in CreateOptions

Reply:
The options should also hold fields for flags based on the `oc` pattern. This can be a little weird at times but this also avoids polluting the package space with variables that are only used for a specific command.

Reply:
Ok fixed now...